### PR TITLE
Fixed bug when setting the same property twice when setting to nulll …

### PR DESCRIPTION
**Pull request by OfirMos**
…at the 2nd time

### CODE REVIEW

File: src/graph/entities/attribute_set.c [hunk 1]
```diff
@@ -233,7 +233,6 @@ bool AttributeSet_UpdateNoClone
 	Attribute_ID attr_id,  // attribute identifier
 	SIValue value          // new value
 ) {
-	ASSERT(set != NULL && *set != NULL);
```
Comment (AviAvni) [inline src/graph/entities/attribute_set.c:236]:
```
ASSERT(set != NUL)
```
is needed

**Review by AviAvni — CHANGES_REQUESTED** [review]

File: tests/flow/test_entity_update.py [hunk 2]
```diff
@@ -534,3 +534,9 @@ def test_36_mix_add_and_remove_node_properties(self):
         self.env.assertEqual(result.properties_set, 1)
         self.env.assertEqual(result.properties_removed, 1)
 
+    def test_36_set_property_null(self):
+        graph.delete()
+        graph.query("CREATE (v{})")
```
Comment (swilly22) [inline tests/flow/test_entity_update.py:539]: "CREATE ()"

File: tests/flow/test_entity_update.py [hunk 3]
```diff
@@ -534,3 +534,9 @@ def test_36_mix_add_and_remove_node_properties(self):
...
+        result = graph.query("MATCH (v{}) SET v.p1 = v.p8, v.p1 = v.p5, v.p2 = v.p4")
```
Comment (swilly22) [inline tests/flow/test_entity_update.py:540]: MATCH (v)

File: src/graph/entities/attribute_set.c [hunk 4]
```diff
@@ -212,10 +211,11 @@ void AttributeSet_Set_Allow_Null
 		return;
 	}
 
+	size_t n = ATTRIBUTESET_BYTE_SIZE(_set) + sizeof(Attribute);
+	_set = rm_realloc(_set, n);
+
 	// allocate room for new attribute
 	_set->attr_count++;
-	size_t n = ATTRIBUTESET_BYTE_SIZE(_set);
-	_set = rm_realloc(_set, n);
```
Comment (swilly22) [inline src/graph/entities/attribute_set.c:218]: Why was this 2 lines moved?

**Review by swilly22 — CHANGES_REQUESTED** [review]

File: src/graph/entities/attribute_set.c (hunk 4 above)
Comment (OfirMos) [reply src/graph/entities/attribute_set.c:218]: @swilly22  Cause _set can be NULL, and doing _set->attr_count is invalid in this case, need first to realloc it.

**Pull request merged**